packages/native/src/useLinking.tsx

```diff
@@ -389,7 +389,7 @@ export default function useLinking(
               // We couldn't find an existing entry to go back to, so we'll go back by the delta
               // This won't be correct if multiple routes were pushed in one go before
               // Usually this shouldn't happen and this is a fallback for that
-              await history.go(historyDelta);
+              await history.push({ path, state });
```

Comment:
This change is not correct. We don't want to always push new screens, it should go forward to back depending on whether the user navigated forward to backward.

Reply:
this is a fallback for the case where the user is navigating to the same screen (e.g. with different route params). it needs to be pushed in order for params to update appropriately.

Reply:
Even for fallback it's not correct. URL doesn't need to be pushed for params to update. It could also replace the params after the url updates.

Reply:
What's the alternative in this situation? I agree that `push` may not be perfect, but `go` is also not perfect and causing other issues right now. The issue @satya164 describes for `push` is that in the fallback scenario if it was actually a replace, we would end up with more routes in browser history than intended (correct?). But at least the correct URL will be shown for the screen they are on, as opposed to what is happening now with `history.go(...)`.

Reply:
This may be a slight tangent, but I feel like we are operating with insufficient knowledge to make the correct changes to the browser history. We only observe the state as it is and try to compare to the previous state without knowing what operations were performed. Although sometimes that's all the knowledge we'd actually have (like when a reset occurs), but most of the time we could know what action was intended (navigate, replace, push, pop, etc.). Would it help if we were listening for actions in the happy case and updating history with the improved knowledge we get from knowing which action was dispatched to effect the state change, then cover edge cases like reset by doing a previous state to current state comparison?

I believe the line mentioned in this change also is the cause for [this issue I submitted](https://github.com/react-navigation/react-navigation/issues/10992) "Multiple Navigation dispatches that include a replace break browser back".